File: parcel-locker-cpp/library/test/DeliveryTest.cpp
//
// Created by student on 08.06.2021.
//

#include <boost/test/unit_test.hpp>
#include  <boost/uuid/random_generator.hpp>
#include "typedefs.h"
#include "model/Delivery.h"
#include "model/Client.h"
#include "model/Parcel.h"
#include "model/List.h"
#include "model/Locker.h"

namespace btt = boost::test_tools;

struct DeliveryFixture
{
    DeliveryPtr delivery1;
    DeliveryPtr delivery2;
    ClientPtr shipper1;
    ClientPtr receiver1;
    PackagePtr box;
    PackagePtr list;
    LockerPtr locker;
    float basePrice = 10;
    boost::uuids::uuid id;

    DeliveryFixture()
    {
        boost::uuids::random_generator gen;
        id = gen();
        locker = std::make_shared<Locker>(20);
        shipper1 =  std::make_shared<Client>("Oscar", "Trel", "321312312");
        receiver1 = std::make_shared<Client>("Bartosh", "Siekan", "123123123");
        delivery1 = std::make_shared<Delivery>(basePrice, id, 10, 20, 30, 10, true, shipper1, receiver1, locker);
        box = std::make_shared<Parcel>(basePrice, 10, 20, 30, 10, true);
        delivery2 = std::make_shared<Delivery>(basePrice, id, true, shipper1, receiver1, locker);
        list = std::make_shared<List>(basePrice, true);
    }
};

BOOST_AUTO_TEST_SUITE(DeliveryTest)

    BOOST_FIXTURE_TEST_CASE(DeliveryParcelConstructorAnGettersConformance, DeliveryFixture)
    {
        BOOST_TEST(delivery1->getId() == id);
        BOOST_TEST(delivery1->getShipper() == shipper1);
        BOOST_TEST(delivery1->getReceiver() == receiver1);
        BOOST_TEST(delivery1->getStatus() == Ready_to_ship);
        BOOST_TEST(delivery1->getAPackage()->getInfo() == box->getInfo());
        BOOST_TEST(delivery1->getLocker() == locker);
    }

    BOOST_FIXTURE_TEST_CASE(DeliveryListConstructorAndGettersConformance, DeliveryFixture)
    {
        BOOST_TEST(delivery2->getId() == id);
        BOOST_TEST(delivery2->getShipper() == shipper1);
        BOOST_TEST(delivery2->getReceiver() == receiver1);
        BOOST_TEST(delivery2->getStatus() == Ready_to_ship);
        BOOST_TEST(delivery2->getAPackage()->getInfo() == list->getInfo());
        BOOST_TEST(delivery2->getLocker() == locker);
    }

    BOOST_FIXTURE_TEST_CASE(DeliverySetterConformance, DeliveryFixture)
    {
        BOOST_TEST(delivery1->getStatus() == Ready_to_ship);
        delivery1->setStatus(Ready_for_pickup);
        BOOST_TEST(delivery1->getStatus() == Ready_for_pickup);
        delivery1->setStatus(Received);
        BOOST_TEST(delivery1->getStatus() == Received);
    }

BOOST_AUTO_TEST_SUITE_END()

File: parcel-locker/src/test/java/pl/pas/parcellocker/model/DeliveryTest.java
package pl.pas.parcellocker.model;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static pl.pas.parcellocker.model.delivery.DeliveryStatus.READY_TO_PICKUP;
import static pl.pas.parcellocker.model.delivery.DeliveryStatus.READY_TO_SHIP;
import static pl.pas.parcellocker.model.delivery.DeliveryStatus.RECEIVED;

import java.math.BigDecimal;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import pl.pas.parcellocker.model.delivery.Delivery;
import pl.pas.parcellocker.model.locker.Locker;
import pl.pas.parcellocker.model.user.Client;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class DeliveryTest {

    private final BigDecimal basePrice = BigDecimal.TEN;
    private Delivery delivery1;
    private Delivery delivery2;
    private Client shipper1;
    private Client receiver1;
    private Locker locker;

    @BeforeAll
    void setup() {
        locker = new Locker("LDZ01", "Gawronska 12, Lodz 12-123", 20);
        shipper1 = new Client("Oscar", "Trel", "321312312");
        receiver1 = new Client("Bartosh", "Siekan", "123123123");
        delivery1 = new Delivery(basePrice, 10, 20, 30, 10, true, shipper1, receiver1, locker);
        delivery2 = new Delivery(basePrice, true, shipper1, receiver1, locker);
    }

    @Test
    void Should_ReturnCorrectObjectsForParcelDelivery_WhenGettersCalled() {
        assertEquals(shipper1, delivery1.getShipper());
        assertEquals(receiver1, delivery1.getReceiver());
        assertEquals(READY_TO_SHIP, delivery1.getStatus());
        assertEquals(locker, delivery1.getLocker());
    }

    @Test
    void Should_ReturnCorrectObjectsForListDelivery_WhenGettersCalled() {
        assertEquals(shipper1, delivery2.getShipper());
        assertEquals(receiver1, delivery2.getReceiver());
        assertEquals(READY_TO_SHIP, delivery2.getStatus());
        assertEquals(locker, delivery2.getLocker());
    }

    @Test
    void Should_SetterSetCorrectStatus() {
        assertEquals(READY_TO_SHIP, delivery1.getStatus());
        delivery1.setStatus(READY_TO_PICKUP);
        assertEquals(READY_TO_PICKUP, delivery1.getStatus());
        delivery1.setStatus(RECEIVED);
        assertEquals(RECEIVED, delivery1.getStatus());
    }
}
